Source/WebCore/rendering/RenderLayer.h

```diff
@@ -1330,7 +1333,9 @@ class RenderLayer final : public CanMakeSingleThreadWeakPtr<RenderLayer>, public
     Markable<ScrollingScope, IntegralMarkableTraits<ScrollingScope, 0>> m_contentsScrollingScope;
 
     std::unique_ptr<TransformationMatrix> m_transform;
-    
+
+    std::optional<LayoutSize> m_snapshottedScrollOffsetForAnchorPositioning;
```

Comment:
I don't see this ever getting cleared, when a layer stops doing anchor positioning. Can you add a test for this?

Reply:
Also, what happens when a scroller appears or disappears between the anchored and anchoring elements?

Reply:
Good point. Added a test and made sure the the offset is cleared.